app/CLI/Application.php

```diff
@@ -30,16 +32,19 @@ public function __construct(KernelInterface $kernel)
 
         $this->setDispatcher($kernel->getApp()->offsetGet('dispatcher'));
 
-        $registeredCommand = array_reduce($kernel->getApp()->keys(), function (&$res, $serviceName) {
-            if ('command.' === substr($serviceName, 0, 8)) {
+        $registeredCommand = array_reduce($kernel->getApp()->keys(), function ($res, $serviceName) use ($commandPrefix) {
+            if ($commandPrefix . '.' === substr($serviceName, 0, 8)) {
                 $res[] = $serviceName;
-
-                return $res;
             }
-        });
+            return $res;
+        }, []);
 
         foreach ($registeredCommand as $serviceName) {
-            $this->add($kernel->getApp()->offsetGet($serviceName));
+            $commandCandidate = $kernel->getApp()->offsetGet($serviceName);
```

Comment:
Just notice that disable the lazy loading of service through the DI, but it's a not real problem.